snowflake-demo/run-model.py

```diff
@@ -0,0 +1,129 @@
+import boto3
+import cloudpickle
+import io
+import logging
+import os
+import pandas as pd
+
+from botocore.client import ClientError
+from flask import Flask
+from flask import jsonify
+from flask import request
+from marshmallow import fields, Schema
+
+app = Flask(__name__)
+
+logger = logging.getLogger()
+logging.basicConfig(
+    format="%(levelname)s [%(asctime)s] %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S"
+)
+
+
+def _get_env_var(var_name: str) -> str:
+    try:
+        return os.environ[var_name]
+    except KeyError:
+        msg = f"Environment variable '{var_name}' must be set"
+        logger.fatal(msg)
+        raise RuntimeError(msg)
+
+MODEL_BUCKET = _get_env_var("MODEL_BUCKET")
+MODEL_FILE = _get_env_var("MODEL_FILE")
+
+logger.info("Fetching model file")
+s3 = boto3.client("s3")
+
+stream = io.BytesIO()
+
+try:
+    logger.info("Downloading model object")
+    s3.download_fileobj(
+        Bucket=MODEL_BUCKET,
+        Key=MODEL_FILE,
+        Fileobj=stream
+    )
+    logger.info("Successfully downloaded model object")
+except ClientError as err:
+    msg = "Failed to download model: {}".format(err)
+    logger.fatal(msg)
+    raise RuntimeError(msg)
+
+logger.info("Deserializing model")
+stream.seek(0)
+model = cloudpickle.loads(stream.read())
+del stream
+logger.info("Successfully deserialized model")
+
+
+class InputSchema(Schema):
+    """
+    Expected schema for input records. In the future, this
+    might be stored along with the model and retrieved at runtime.
+
+    For details on timestamp fields in marshmallow, see
+    https://marshmallow.readthedocs.io/en/stable/_modules/marshmallow/fields.html
+    """
+    passenger_count = fields.Int(required=True)
+    tpep_pickup_datetime = fields.DateTime(required=True, format="iso8601")
```

Comment:
in the parquet files for the demo, the `tpep_` part has been removed, so its just `pickup_datetime` and `dropoff_datetime`, but im not sure it matters here?

Reply:
yeah it's irrelevant because the time here is being provided by front-end code. I can make it whatever that code wants to use.

Also `dropoff_datetime` is irrelevant for our case, right?

Reply:
yeah `dropoff_datetime` not needed here, just said that for awareness